src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -758,10 +758,23 @@ void executeSuppressionActions(IEnumerable<Diagnostic> reportedDiagnostics, bool
                 var suppressors = this.Analyzers.OfType<DiagnosticSuppressor>();
                 if (concurrent)
                 {
-                    Parallel.ForEach(suppressors, suppressor =>
+                    try
                     {
-                        AnalyzerExecutor.ExecuteSuppressionAction(suppressor, getSuppressableDiagnostics(suppressor));
-                    });
+                        Parallel.ForEach(suppressors, suppressor =>
+                        {
+                            AnalyzerExecutor.ExecuteSuppressionAction(suppressor, getSuppressableDiagnostics(suppressor));
+                        });
+                    }
+                    catch (AggregateException ex)
+                    {
+                        // Unwrap and throw the inner OperationCanceledException.
+                        if (ex.InnerExceptions.Count == 1 && ex.InnerException is OperationCanceledException)
```

Comment:
You'll want to throw `OperationCanceledException` even if more than one was cancelled, as long as all are.